test/framework/cluster_proxy.go

```diff
@@ -188,12 +188,14 @@ type clusterProxy struct {
 }
 
 // NewClusterProxy returns a clusterProxy given a KubeconfigPath and the scheme defining the types hosted in the cluster.
```

Comment:
Q: What about all the placers where we are using GetKubeconfigPath() today and where we are expecting to actually get a kubeconfig path?

Reply:
It is a good question. I have not found its usage inside the cluster_proxy itself, but yes, it is a good point. 
However, it would be the same behaviour - the method would return an empty string and then it is up to the client to decide, if they fall back to InClusterConfig or stop progressing.

It is still better than returning some default path, that would not exist

Reply:
Hm yeah. We have a lot of usages of GetKubeconfigPath() in our e2e tests. Not sure if these tests will still work, but maybe in these cases we always have a kubeconfigPath set?

Reply:
Just to double check. In the in-cluster case with the old code after l.196 kubeconfigPath just stays empty, right?

So then this change is clearly just a step in the right direction

Reply:
Yes, if it is in-cluster - the kubeconfigPath variable remains empty and would return empty in GetKubeconfigPath().


Can I run some more e2e tests if you want and could point me what to run? (or you if you have time).

Reply:
Okay, then I'll think we're good. I'll trigger the tests